socialhome/streams/app/components/Stream.vue

```diff
@@ -1,30 +1,27 @@
 <template>
     <div>
         <div class="container-flex">
-            <div v-show="$store.state.stream.hasNewContent" class="new-content-container">
+            <div v-show="$store.state.hasNewContent" class="new-content-container">
                 <b-button @click.prenvent.stop="onNewContentClick" variant="link" class="new-content-load-link">
                     <b-badge pill variant="primary">
-                        {{ $store.state.stream.newContentLengh }} new posts available
+                        {{ $store.state.newContentLengh }} new posts available
                     </b-badge>
                 </b-button>
             </div>
             <div v-images-loaded.on.progress="onImageLoad" v-masonry v-bind="masonryOptions">
                 <div class="stamped">
-                    <stamped-element></stamped-element>
+                    <stamped-element />
                 </div>
                 <div class="grid-sizer"></div>
                 <div class="gutter-sizer"></div>
                 <stream-element
                     class="grid-item"
-                    v-for="content in $store.state.contentList"
+                    v-for="content in $store.getters.contentList"
```

Comment:
I'm wondering, won't any change in any content always re-render every single content instead of one of them? Any idea? Also does Vue have to constantly check the getter whether there are new items in the list? I really have no idea how it should work, just wondering out aloud.

We need only one content to re-render when it changes or new appears, otherwise we don't get any benefit from using Vue. Currently the stream re-renders everything on one change.

Reply:
According to the documentation Vuex' getters are similar to component's computed properties which are cached. As long as the base data don't change, the list don't change. But I suppose the list is entirely recomputed when an element is inserted to or removed from the original list. Which can be heavy, yes.

Good thing is that everything is async in Vue. The view is rerendered the next tick after the property was entirely recomputed so that shouldn't slower the stream in any way.

As for the view re-render, Vue only renders what changed thanks to virtual DOM. This is why list rendering requires [the `id` atribute](https://github.com/jaywink/socialhome/pull/273/files/c8a6be58b6883dfafa3b492096834bf3e1925a25#diff-f6bf769c688863e153cb8f8d589a3803R22), so that only new or removed elements change.

[There's an example closed to that in Vuex documentation](https://vuex.vuejs.org/en/getters.html). You can read about the in-place patch strategy [here](https://vuejs.org/v2/guide/list.html#key).

Reply:
OK, good enough for me! 👍